### ISSUE TITLE
🐛 Theme doesn't match theme preview

### ISSUE THREAD

**Issue by casey** [dandavison/delta#596]
Thanks for this awesome tool!

I'm trying to set my theme, and the output of `git diff` doesn't look like `delta --show-syntax-themes`.

In this screenshot, my `~/.config/git/config` is on the left, the output of `delta --show-syntax-themes` is on the upper right, and the output of `git diff` is on the lower right.

<img width="1130" alt="Screen Shot 2021-05-09 at 8 19 17 PM" src="https://user-images.githubusercontent.com/1945/117601733-e8948d80-b103-11eb-80b1-b16637ca5f32.png">

In particular, this text was added:

```

[delta]
syntax-theme = 1337
```

So I was expecting it to have a green background like the output of `delta --show-syntax-themes`.

I'm using alacritty with the following colors:

```
dark:
  primary:
    background: '0x151718'
    foreground: '0xd6d6d6'
  cursor:
    text: '0x151718'
    cursor: '0xd6d6d6'
  normal:
    black:   '0x151718'
    red:     '0xcd3f45'
    green:   '0x9fca56'
    yellow:  '0xe6cd69'
    blue:    '0x55b5db'
    magenta: '0xa074c4'
    cyan:    '0x55dbbe'
    white:   '0xd6d6d6'
  bright:
    black:   '0x41535b'
    red:     '0xdb7b55'
    green:   '0x282a2b'
    yellow:  '0x3b758c'
    blue:    '0x43a5d5'
    magenta: '0xeeeeee'
    cyan:    '0x8a553f'
    white:   '0xffffff'
```

And base16-seti in the shell.

**Comment by dandavison** [comment]
Hi @casey, sorry to be slow here. Did you get to the bottom of this? The answer didn't immediately jump out to me, but we can definitely get to the bottom of it. I agree that your `git diff` output on the markdown file seems wrong -- it should have background colors. That suggests possibly something not handling a 24-bit color code. There are a lot of possible things to investigate:

- Does it work in a different terminal emulator?
- Are you using tmux?
- Does a 24-bit color test look correct in your terminal (see https://github.com/dandavison/delta#24-bit-color-truecolor)
- Does it work with another syntax theme? What about `syntax-theme=ansi`?
- Does `delta --24-bit-color=never` help?
- Is it a problem with the pager (less)? You can use `delta --paging=never` to prevent delta from, sending output to a pager (but `--show-syntax-themes` uses the pager also so unclear how this explains your observations)

**Comment by casey** [comment]
Hey @dandavison, no worries at all!

I should have done more debugging before reporting this, but I was surprised with how the output was different between `--show-syntax-themes` and `git diff`.

I did some digging, and the problem is that 24-bit color wasn't working properly. I'm using the Alacritty terminal with Tmux. I got it working by adding this to `~/.tmux.conf`:

```
set -ga terminal-overrides ",xterm-256color:Tc"
```

That fixed it, so I'm going to go ahead and close this.

One thing that I thought of that might cut down on time spent on these kinds of issues is a built-in terminal test for `delta`. Something like `delta --terminal-test`, which would have output that could be compared against a canonical version to see if the terminal was configured correctly.

It might output a grid with a bunch of colors that only appears correctly if 24-bit color is enabled. (Ideally it would also exercise any other terminal features that delta uses.)

Sort of like a [CSS Acid Test](https://en.wikipedia.org/wiki/Acid3), but for the terminal. This would be so useful, that it might be worth making it a standalone project.

**Comment by dandavison** [comment]
Ok excellent, glad you got it fixed, I also use tmux and had problems with 24-bit color initially and recall spending some time before coming across that (somewhat opaque!) fix. I should have thought of it earlier.

Good call, `delta --terminal-test` (or `delta --doctor`?) would be very helpful. I made a very tentative start on that with this shell-script: [delta:etc/bin/diagnostics](https://github.com/dandavison/delta/blob/master/etc/bin/diagnostics). (I think it would help Windows users in particular because there the terminal emulator capabilities seem a bit more patchy and there also often seem to be problems with old less versions being installed.) I've opened https://github.com/dandavison/delta/issues/618 for this.
